.github/workflows/phpunit.yml

```
@@ -0,0 +1,99 @@
+name: PHPUnit tests
+
+on: pull_request
+
+env:
+  WP_TRAVISCI: "phpunit"
+  COMPOSER_ROOT_VERSION: "dev-master"
+
+jobs:
+  phpunit:
+    name: "PHP: ${{ matrix.php }} WP: ${{ matrix.wp }}"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        php: ["5.6", "7.0", "7.2", "7.3", "7.4snapshot"]
+        wp: ["master", "latest", "previous"]
+        exclude:
+        # excludes node 4 on macOS
+        - php: "7.0"
+        - php: "7.2"
+        - wp: "latest"
```

Lines 20–22 don't do what you probably intend:


* Line 20 excludes any job where php is "7.0", so it's the same as removing "7.0" from line 16.
* Line 21 does the same for 7.2.
* Line 22 excludes any job where wp is "latest", so it's the same as removing "latest" from line 17.

I think what you might intend is something like this:
```suggestion
    continue-on-error: ${{ matrix.experimental }}

    strategy:
      fail-fast: false
      matrix:
        # Run tests against current WP in all supported PHP versions.
        php: [ "5.6", "7.0", "7.2", "7.3", "7.4" ]
        wp: [ "latest" ]
        experimental: [ false ]

        include:
          # Run tests against WP previous and master in one PHP version.
          - php: "7.4"
            wp: "master"
          - php: "7.4"
            wp: "previous"

          # Experimental PHP 8 tests
          - php: "8.0"
            wp: "latest"
            experimental: true
```

---

My plan was to add conditionals to these `exclude` clauses. I was thinking to exclude them for PRs, and run them only for the `master` branch. But I like your suggestion